examples/src/keyring/aws_kms/act_like_aws_kms_master_key_provider.py

```diff
@@ -0,0 +1,88 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""
+Before there were keyrings, there were master key providers.
+Master key providers were the original configuration structure
+that we provided for defining how you want to protect your data keys.
+
+The AWS KMS master key provider was the tool that we provided for interacting with AWS KMS.
+Like the AWS KMS keyring,
+the AWS KMS master key provider encrypts with all CMKs that you identify,
+but unlike the AWS KMS keyring,
+the AWS KMS master key provider always attempts to decrypt
+*any* data keys that were encrypted under an AWS KMS CMK.
+We have found that separating these two behaviors out,
+as we did in the AWS KMS keyring,
+makes it more clear what behavior to expect.
+However, as you migrate away from master key providers to keyrings,
+you might need to replicate the exact behavior of the AWS KMS master key provider.
```

Comment:
maybe remove "exact", since its not quite exactly the same behavior in certain edge cases

Reply:
```suggestion
you might need to replicate the behavior of the AWS KMS master key provider.
```